### PR TITLE
[bitnami/cassandra] First version

### PR DESCRIPTION
**Pull request by javsalgar**

### CODE REVIEW

File: bitnami/cassandra/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: cassandra
+version: 1.0.0
```
Comment (carrodher) [inline bitnami/cassandra/Chart.yaml]: I think we have initialized other charts with `0.0.1`

File: bitnami/cassandra/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+The above parameters map to the env variables defined in [bitnami/cassandra](http://github.com/bitnami/bitnami-docker-cassandra). For more information please refer to the [bitnami/cassandra](http://github.com/bitnami/bitnami-docker-cassandra) image documentation.
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install --name my-release \
+  --set cassandraUser=admin,cassandraPassword=password\
+    ./cassandra
+```
+
```
Comment (carrodher) [inline bitnami/cassandra/README.md]: Extra line

File: bitnami/cassandra/values.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  pullPolicy: Always
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
+
+## Specify a service type
+## ref: http://kubernetes.io/docs/user-guide/services/
```
Comment (carrodher) [inline bitnami/cassandra/values.yaml]: We usually add a line with `##` between the description and the code, something like:
```
## Bitnami Cassandra image version
## ref: https://hub.docker.com/r/bitnami/cassandra/tags/
##
image:
  registry: docker.io
  repository: bitnami/cassandra
```

Can you check other cases?

File: bitnami/cassandra/Chart.yaml (hunk 1 above)
Comment (juan131) [reply bitnami/cassandra/Chart.yaml]: +1

File: bitnami/cassandra/README.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
+# cassandra
+
+[cassandra](https://cassandra.apache.org) Apache Cassandra is a free and open-source distributed database management system designed to handle large amounts of data across many commodity servers or datacenters.
+
+## TL;DR;
+
+```console
+$ helm install .
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use:

```
helm repo add bitnami https://charts.bitnami.com/bitnami
helm install bitnami/cassandra
```

File: bitnami/cassandra/README.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+- Kubernetes 1.8+
+- PV provisioner support in the underlying infrastructure
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```console
+$ helm install --name my-release ./cassandra
```
Comment (juan131) [inline bitnami/cassandra/README.md]: I'd include the complete name of the cart on every instruction.
```
$ helm install --name my-release bitnami/cassandra
```

File: bitnami/cassandra/README.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+The command deploys one node with Cassandra on the Kubernetes cluster in the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+
+> **Tip**: List all releases using `helm list`
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` petset:
```
Comment (juan131) [inline bitnami/cassandra/README.md]: `petset` ??

File: bitnami/cassandra/README.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the cassandra chart and their default values.
+
+| Parameter                                  | Description                                                                                                    | Default                                              |
+|--------------------------------------------|----------------------------------------------------------------------------------------------------------------|------------------------------------------------------|
+| `global.imageRegistry`                     | Global Docker image registry                                                                                   | `nil`                                                |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Be consistent. You used `Global Docker image registry`, then `Cassandra Image registry`

Choose between `image` or `Image` and apply everywhere

File: bitnami/cassandra/README.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+| `image.registry`                           | Cassandra Image registry                                                                                           | `docker.io`                                          |
+| `image.repository`                         | Cassandra Image name                                                                                               | `bitnami/cassandra`                                      |
+| `image.tag`                                | Cassandra Image tag                                                                                                | `3.11.3`                                          |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use `{VERSION}` instead of hardcoding `3.11.3`

File: bitnami/cassandra/README.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+| `service.type`                       | Kubernetes Service type                                                                          | `ClusterIP`                                          |
+| `service.nodePort`                   | Kubernetes Service nodePort                                                                      | `nil`                                                |
+| `service.loadBalancerIP`             | LoadBalancerIP if service type is `LoadBalancer`                                                   | `nil`                            | 
+| `service.annotations`                | Annotations for the service                                                                            | {}                                                   |
+| `persistence.enabled`               | Use PVCs to persist data                                                                        | `true`                                               |
+| `persistence.storageClass`          | Storage class of backing PVC                                                                                   | `generic`                                            |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use `Persistent Volume Storage Class` for Description

File: bitnami/cassandra/README.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+| `persistence.annotations`                | Annotations for the PVC                                                                            | {}                                                   |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use `Persistent Volume Claim annotations` for Description

File: bitnami/cassandra/README.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+| `persistence.accessModes`           | Persistent Volume Access Modes                                                                                 | `[ReadWriteOnce]`                                    |
+| `persistence.size`                  | Size of data volume                                                                                            | `8Gi`                                                |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use `Persistent Volume size` for Description

File: bitnami/cassandra/README.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+| `resources`                         | CPU/Memory resource requests/limits                                                               |  `{}`                         |
+| `cluster.name`                         | Cassandra cluster name                                                               |  `cassandra`                         |
+| `cluster.replicaCount`                         | Number of cassandra nodes                                                               |  `1`                         |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: `cassandra` -> `Cassandra`

File: bitnami/cassandra/README.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+| `cluster.seedCount`                           | Number of seed nodes (note: must be greater or equal than 1 and less or equal to `cluster.replicaCount`)                                                   | `1`                                                |
+| `dbUser.user`                           | Cassandra admin user                                                   | `cassandra`                                                |
+| `dbUser.forcePassword`                           | Force the user to provide a non-empty password for `dbUser.user`                                                   | `false`                                                |
+| `dbUser.password`                              | Password for `dbUser.user`. Randomly generated if empty                                                                                                   | (Random generated)                                               |
+| `dbUser.existingSecret`                                 | Use an existing secret object for `dbUser.user` password (will ignore `dbUser.password`)                                                                 | ``                                   |
```
Comment (juan131) [inline bitnami/cassandra/README.md]: Use `nil` for Default

File: bitnami/cassandra/templates/NOTES.txt [hunk 14]
```diff
@@ -0,0 +1,59 @@
+** Please be patient while the chart is being deployed **
+
+Cassandra can be accessed through the following urls from within the cluster:
```
Comment (juan131) [inline bitnami/cassandra/templates/NOTES.txt]: I'd use `URLs` instead

File: bitnami/cassandra/templates/NOTES.txt [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+   
+   cqlsh -u {{ .Values.dbUser.user }} -p $CASSANDRA_PASSWORD $NODE_IP $NODE_PORT
+
+{{- else if contains "LoadBalancer" .Values.service.type }}
+
+  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+        Watch the status with: 'kubectl get svc --namespace {{ .Release.Namespace }} -w {{ template "redis.fullname" . }}'
+
+    export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "redis.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+   cqlsh -u {{ .Values.dbUser.user }} -p $CASSANDRA_PASSWORD $SERVICE_IP
```
Comment (juan131) [inline bitnami/cassandra/templates/NOTES.txt]: Is it well indented?

File: bitnami/cassandra/templates/NOTES.txt [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+{{- else if contains "ClusterIP" .Values.service.type }}
+
+    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ template "cassandra.fullname" . }} {{ .Values.service.cqlPort }}:{{ .Values.service.cqlPort }} &
+      cqlsh -u {{ .Values.dbUser.user }} -p $CASSANDRA_PASSWORD 127.0.0.1 {{ .Values.service.cqlPort }}
```
Comment (juan131) [inline bitnami/cassandra/templates/NOTES.txt]: Is it well indented?

File: bitnami/cassandra/templates/_helpers.tpl [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+{{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+{{- end -}}
+{{- end -}}
+
+
+{{/*
+Return the proper metrics image name
+*/}}
+{{- define "metrics.image" -}}
```
Comment (juan131) [inline bitnami/cassandra/templates/_helpers.tpl]: Use `cassandra.metrics.image`

Let's say you use Cassandra with KONG and you have another Prometheus exporter. There'd be a "namespace conflict" if both uses a macro which name `metrics.image`

File: bitnami/cassandra/templates/cassandra-secret.yaml [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+  name: {{ template "cassandra.fullname" . }}
+  labels:
+    app: "{{ template "cassandra.name" . }}"
+    chart: {{ template "cassandra.chart" . }}
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+type: Opaque
+data:
+  {{- if .Values.dbUser.password }}
+  cassandra-password: "{{ .Values.dbUser.password | b64enc }}"
```
Comment (juan131) [inline bitnami/cassandra/templates/cassandra-secret.yaml]: To be consistent with the format you used on labels, use:

```
{{ .Values.dbUser.password | b64enc | quote }}
```

File: bitnami/cassandra/templates/headless-svc.yaml [hunk 19]
```diff
@@ -0,0 +1,31 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "cassandra.fullname" . }}-headless
+  labels:
```
Comment (juan131) [inline bitnami/cassandra/templates/headless-svc.yaml]: Be consistent with labels format. Use the same one you used on the secret definition.

File: bitnami/cassandra/templates/networkPolicy.yaml [hunk 20]
```diff
@@ -0,0 +1,31 @@
+{{- if .Values.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: {{ template "networkPolicy.apiVersion" . }}
+metadata:
+  name: "{{ template "cassandra.fullname" . }}"
+  labels:
```
Comment (juan131) [inline bitnami/cassandra/templates/networkPolicy.yaml]: Same concern about the format

File: bitnami/cassandra/templates/networkPolicy.yaml [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    heritage: "{{ .Release.Service }}"
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "cassandra.name" . }}
+      release: "{{ .Release.Name }}"
+  ingress:
+    # Allow inbound connections
+    - ports:
+      - port: 6379
```
Comment (juan131) [inline bitnami/cassandra/templates/networkPolicy.yaml]: Hardcode?

File: bitnami/cassandra/templates/service.yaml [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+{{- if .Values.service.nodePort }}
+  nodePort: {{ .Values.service.nodePort }}
+{{- end }}
+{{ if eq .Values.service.type "LoadBalancer" -}} {{ if .Values.service.loadBalancerIP -}}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
+{{ end -}}
+  type: {{ .Values.service.type }}
+  ports:
+  - name: cql
+    port: {{ default 9042 .Values.service.cqlPort }}
```
Comment (juan131) [inline bitnami/cassandra/templates/service.yaml]: Do not use "default". Default values should come from **values.yaml**

**Pull request merged**